core/src/test/java/org/apache/shiro/session/mgt/ExecutorServiceSessionValidationSchedulerTest.java

```diff
@@ -0,0 +1,106 @@
+        executorServiceSessionValidationScheduler.setThreadNamePrefix("test-");
+        executorServiceSessionValidationScheduler.setInterval(1000L);
+        executorServiceSessionValidationScheduler.enableSessionValidation();
+    }
+
+    @Test
+    public void timeoutSessionValidate() throws InterruptedException {
+        Session session = new SimpleSession();
+        session.setTimeout(2000L);
+        defaultSessionManager.create(session);
+        Thread.sleep(5000L);
+        Assert.assertTrue(defaultSessionManager.getActiveSessions().isEmpty());
+        Assert.assertTrue(executorServiceSessionValidationScheduler.isEnabled());
+    }
+
+    @Test
+    public void stopSessionValidate() throws InterruptedException {
+        Session session = new SimpleSession();
+        session.setTimeout(10000L);
+        defaultSessionManager.create(session);
+        Thread.sleep(1000L);
+        session.stop();
+        Thread.sleep(3000L);
+        Assert.assertTrue(defaultSessionManager.getActiveSessions().isEmpty());
+        Assert.assertTrue(executorServiceSessionValidationScheduler.isEnabled());
+    }
+
+    @Test
+    public void enableSessionValidation() throws InterruptedException {
+        Assert.assertTrue(executorServiceSessionValidationScheduler.isEnabled());
+        executorServiceSessionValidationScheduler.disableSessionValidation();
+        Thread.sleep(2000L);
+        Assert.assertFalse(executorServiceSessionValidationScheduler.isEnabled());
+        executorServiceSessionValidationScheduler.enableSessionValidation();
+        Thread.sleep(2000L);
+        Assert.assertTrue(executorServiceSessionValidationScheduler.isEnabled());
+    }
+
+    @Test
+    public void threadException() throws InterruptedException {
+        Session session = new SimpleSession();
+        defaultSessionManager = new FakeDefaultSessionManager();
+        defaultSessionManager.setDeleteInvalidSessions(true);
+        executorServiceSessionValidationScheduler = new ExecutorServiceSessionValidationScheduler();
+        executorServiceSessionValidationScheduler.setSessionManager(defaultSessionManager);
+        executorServiceSessionValidationScheduler.setThreadNamePrefix("test-");
+        executorServiceSessionValidationScheduler.setInterval(1000L);
+        executorServiceSessionValidationScheduler.enableSessionValidation();
+        defaultSessionManager.create(session);
+        Thread.sleep(2000L);
+        session.stop();
+        Thread.sleep(2000L);
+        Assert.assertFalse(defaultSessionManager.getActiveSessions().isEmpty());
+        Assert.assertTrue(executorServiceSessionValidationScheduler.isEnabled());
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        executorServiceSessionValidationScheduler.disableSessionValidation();
+    }
+
+    private class FakeDefaultSessionManager extends DefaultSessionManager {
+        @Override
+        public void validateSessions() {
+            Byte[] fake = null;
```

Comment:
Maybe just throw an exception here to make this more clear?

Reply:
@bdemers Done ;)